code/modules/tgui_panel/tgui_panel.dm

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2020 Aleksej Komarov
+ * SPDX-License-Identifier: MIT
+ */
+
+/**
+ * tgui_panel datum
+ * Hosts tgchat and other nice features.
+ */
+/datum/tgui_panel
+	var/client/client
+	var/datum/tgui_window/window
+	var/broken = FALSE
+	var/initialized_at
+
+/datum/tgui_panel/New(client/client)
+	src.client = client
+	window = new(client, "tgui-panel")
```

Comment:
Maybe allow the panel to have an ID passed to it. It could come in handy when a second panel is required, for example for a tgui stat panel, unless I'm missing something.

https://github.com/tgstation/tgstation/pull/50963, for reference.

Reply:
The plan is to let that panel consume the whole right part of the screen, and in that case stat panel would be inside tgui-panel and stat subsystem would send messages to tgui-panel.

If it has to be a separate panel/window, you can do it with /datum/tgui_window, which is a general purpose abstraction. It does not require a lot of code to set up.

Reply:
ah ok, thank you.